plugin/cancel/README.md

```diff
@@ -0,0 +1,34 @@
+# cancel
+
+## Name
+
+*cancel* - a plugin that cancels a request's context after 5001 milliseconds.
+
+## Description
+
+The *cancel* plugin creates a canceling context for each request. It adds a timeout that gets
+triggered after 5001 milliseconds.
+
+The 5001 number is chosen because the default timeout for DNS clients is 5 seconds, after that they
+give up.
+
+A pluging interested in the cancelation status should call `plugin.Done()` on the context. If the
```

Comment:
Should this be plumbed through the higher level plugin.NextOrFailure? Or just a best practice, in which case we should update other plugins to use it.